### PR TITLE
Kotlin 1.7.10

### PR DESCRIPTION
**Pull request by 3flex**
https://blog.jetbrains.com/kotlin/2022/06/kotlin-1-7-0-released/

https://github.com/JetBrains/kotlin/releases/tag/v1.7.0

### CODE REVIEW

**Comment by BraisGabin** [comment]
Could you explain a bit the changes you did on the tests? I don't untesrtand why those changes are needed.

**Comment by 3flex** [comment]
Odd failure with binary-compatibility-validator:
```
> Incompatible version of Kotlin metadata.
  Maximal supported Kotlin metadata version: 1.5.1,
  io/github/detekt/psi/FilePath Kotlin metadata version: 1.7.1.
  As a workaround, it is possible to manually update 'kotlinx-metadata-jvm' version in your project.
```
This is strange because I'd expect a similar error using 1.6.21, as that uses metadata version 1.6.0. So I'm really not sure what the root cause is.

**Comment by cortinico** [comment]
Unless strictly necessary, I would merge this after we release 1.21 so it will go in 1.22. Opinion?

**Comment by 3flex** [comment]
> Could you explain a bit the changes you did on the tests?

I've added a reference to each commit in this PR, which links to the "What's new in Kotlin 1.7.0" page, the Kotlin 1.7 compatibility guide, or the individual commit in JetBrains/kotlin as appropriate. Hopefully this provides enough context for each change.

**Review by BraisGabin — APPROVED** [review]
Thanks for split the PR and all those links! Really easy to follow.

**Comment by 3flex** [comment]
Agree - this should be tracked in a separate issue.

**Comment by chao2zhang** [comment]
Do we need to update `module.gradle.kts` - `kotlinOptions.apiVersion`? The existing value 1.4 is already deprecated since Kotlin 1.6, and I believe we need to bump this further.

**Review by chao2zhang — APPROVED** [review]

**Comment by 3flex** [comment]
It's set to 1.4 to ensure compatibility with Gradle 6.x which sets the same version.

There's probably a way to bump it up for other modules and just hold it back for the Gradle plugin but I haven't looked into it.

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [versions]
 dokka = "1.6.21"
 jacoco = "0.8.8"
-kotlin = "1.6.21"
+kotlin = "1.7.0"
```
Comment (schalkms) [inline gradle/libs.versions.toml:4]: Should be 1.7.10 then.

```suggestion
kotlin = "1.7.10"
```

**Pull request merged**